serialisable.hpp

```diff
@@ -104,6 +108,23 @@ class Serialisable {
 		inline virtual double& getDouble() {
 			return value_;
 		}
+		inline void write(std::ostream& out, int = 0) {
+			out << value_;
+			double whole;
+			if (std::modf(value_, &whole) == 0)
+				out << ".0";
+		}
+	};
+	struct JSONint : public JSON {
+		int64_t value_;
```

Comment:
Both `long long int` and `unsigned long long int` are managed. The serialisation will be signed (e.g. `MAX_UINT64` will serialise to `-1`, but deserialisation will yield the correct value due to wrap-around behaviours.